library/std/src/os/mod.rs

```diff
@@ -23,6 +23,7 @@ pub use crate::sys::windows_ext as windows;
 pub mod linux;
 
 #[cfg(doc)]
+#[stable(feature = "rust1", since = "1.0.0")]
 pub use crate::sys::wasi_ext as wasi;
```

Comment:
This is weird. Why is this doc-only? Why would a stability attribute be needed if it's not present when running normally?

Reply:
The attribute needs to be applied to the item since we're now checking for it. As for why "doc-only", I think this question is out of the scope of this PR. ;)

Reply:
@sunfishcode do you know why this is `cfg(doc)` only? It seems weird to have stability attributes on things that aren't available when writing your own code.

Reply:
Also @GuillaumeGomez 1.0.0 is wrong I think, wasi was first added in https://github.com/rust-lang/rust/pull/59464 which landed in 1.35.0.

Reply:
The reason this is `cfg(doc)` is the same reason the `unix`, and `windows`, modules are `cfg(doc)`.

My understanding of this code is from the comment above, on line 6. This is all a way to include the platform-specific documentation for all the platforms in a single documentation build. These aren't the definitions that get used when doing actual non-doc compilation.

Reply:
Got it, thanks. Is 1.35 the right date for being marked as stable? I think @alexcrichton mentioned in https://github.com/rust-lang/rust/pull/59464 that the module wasn't being stabilized right away, but I couldn't find a PR stabilizing it anywhere.

Reply:
I used the same as https://github.com/rust-lang/rust/blob/master/library/std/src/sys/mod.rs#L114

Reply:
That stability attribute also seems wrong.

Reply:
Yup, but not sure how to handle this situation...

Reply:
The individual inner modules such as [`io`](https://github.com/rust-lang/rust/blob/master/library/std/src/sys/wasi/ext/io.rs#L4) and [`fs`](https://github.com/rust-lang/rust/blob/master/library/std/src/sys/wasi/ext/fs.rs#L4) have `#![unstable(feature = "wasi_ext", issue = "none")]`. So I'd imagine it's ok to mark the outer module as `stable`, because that just contains the inner modules, and the fact that there just exists a module inside it named "fs" and one named "io" is something I think we can consider stable.

Reply:
Ok, then @GuillaumeGomez can you update the attributes to show 1.35? I think this is good to go after that :)